api/v1beta1/azuremanagedmachinepool_webhook.go

```diff
@@ -174,6 +174,13 @@ func (mw *azureManagedMachinePoolWebhook) ValidateUpdate(ctx context.Context, ol
 		allErrs = append(allErrs, err)
 	}
 
+	if err := webhookutils.ValidateImmutable(
```

Comment:
Could you add a test for this block in [this file](https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/1ab9b7448594f5466a1c74b95696cf9ce88adea3/api/v1beta1/azuremanagedmachinepool_webhook_test.go#L546-L559) as well?

Reply:
done.